lib/excon/socket.rb

```diff
@@ -40,12 +40,14 @@ def params=(new_params)
 
     def_delegators(:@socket, :close)
 
+
     def initialize(data = {})
       @data = data
       @nonblock = data[:nonblock]
       @port ||= @data[:port] || 80
       @read_buffer = String.new
       @eof = false
+      @backend_eof = false
```

Comment:
Just to confirm I'm understanding this. It looks like the distinction this introduces is that `@backend_eof` is when the socket raises that it is done, whereas `@eof` is when that has occured AND the `@read_buffer` is also empty. Is that correct? I wanted to make sure, since this distinction didn't exist previously, or at least not in this way.

Reply:
Yes, that's right. We need some way to distinguish "we've fully read the underlying socket" from "we have no more data to return to the client"

Reply:
Got it, thanks for the clarification.